packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const baseClass = 'search-bar';
+const noop = () => {};
+
+class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.onChangeDebounced = debounce(this.props.debounceInMs, value => {
+      this.props.onChange(value);
+    });
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: this.props.compact
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.onChangeDebounced(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
```

Comment:
what if `event.preventDefault();` will be needed?:)

Reply:
Can you tell me more about it? Because right now I'm confused if I should use `e.preventDefault()` in handleKeyPress function or use it based on some param

Reply:
hmm, let's add optional property `onKeyDown`, if someone will need to call `preventDefault` method of event it will be possible,
here you will need to add the following code:
```
if (this.props.onKeyDown) {
  this.props.onKeyDown(event)
}
```